File: Math/Prime/python/prime.py
#!/usr/bin/python
import sys
import math

def is_prime(number):
    if number <= 1:
        return False
    if number == 2:
        return True
    if number % 2 == 0:
        return False

    for i in range(3, 1 + int(math.sqrt(number)), 2):
        if (number % i) == 0:
            return False
    return True

number = input("Enter a number : ")

if is_prime(int(number)):
    print '{} is a prime number'.format(number)
else:
    print '{} is not a prime number'.format(number)

File: Math/Prime/ruby/prime.rb
def is_prime(number)
    if (number <= 1)
        return false
    end
    (2..Math.sqrt(number).to_i).each do |i|
        if ((number % i) == 0)
            return false
        end
    end
    return true
end

puts "Enter a number : "
number = gets.chomp.to_i

if is_prime(number)
    print "%d is a prime number\n" % number
else
    print "%d is not a prime number\n" % number
end
